### PR TITLE
Far grab rolls up parent tree when grabbing and add offset to LaserPointer::lockEndUUID

### PR DESCRIPTION
**Pull request by druiz17**
This PR fixes issues relating to a Far grab does not roll up parent hierarchy when grabbing and far-grab grabs center of entity rather than where the beam hit

Tickets to these issuues
- https://highfidelity.fogbugz.com/f/cases/8073
- https://highfidelity.fogbugz.com/f/cases/8920/Far-grab-does-not-roll-up-parent-hierarchy-when-grabbing

### CODE REVIEW

**Comment by druiz17** [comment]
Test plan 
- create a set of dynamic and non-dynamic entities that are parented to each other
- far grab the parent or child entities with each set
- grabbing parent entity should move the whole set
- grabbing child entities should also move the whole set
- when far grabbing the pink laser should lock on where you interested the entity, it should not lock onto the center of the entity

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4712/HighFidelity-Beta-PR11760-3b15897b123e40d6faa1dab4edb46bc36f77fd04.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4712/HighFidelity-Beta-PR11760-3b15897b123e40d6faa1dab4edb46bc36f77fd04.exe

**Comment by Nex-Pro** [comment]
Testing

File: interface/src/raypick/LaserPointer.cpp [hunk 1]
```diff
@@ -25,7 +25,7 @@ LaserPointer::LaserPointer(const QVariant& rayProps, const RenderStateMap& rende
     _distanceScaleEnd(distanceScaleEnd),
     _rayPickUID(DependencyManager::get<RayPickScriptingInterface>()->createRayPick(rayProps))
 {
-    
+    _offsetMat = glm::mat4();
```
Comment (SamGondelman) [inline interface/src/raypick/LaserPointer.cpp]: This variable was deleted.

File: interface/src/raypick/LaserPointer.cpp [hunk 2]
```diff
@@ -233,9 +235,11 @@ void LaserPointer::setLaserLength(const float laserLength) {
     });
 }
 
-void LaserPointer::setLockEndUUID(QUuid objectID, const bool isOverlay) {
+void LaserPointer::setLockEndUUID(QUuid objectID, const bool isOverlay, const glm::mat4& offsetMat) {
     withWriteLock([&] {
-        _objectLockEnd = std::pair<QUuid, bool>(objectID, isOverlay);
+        _lockEndObject.id = objectID;
+        _lockEndObject.isOverlay;
```
Comment (SamGondelman) [inline interface/src/raypick/LaserPointer.cpp]: Need to set isOverlay

**Comment by druiz17** [comment]
@Nex-Pro I made some other changes. Can you wait for the new build to finish before testing.

File: scripts/system/controllers/controllerModules/farActionGrabEntity.js [hunk 3]
```diff
@@ -113,18 +114,71 @@ Script.include("/~/system/libraries/controllers.js");
...
+                this.previousCollisionStatus = this.targetEntityProps.collisionless;
+                Entities.editEntity(this.targetEntityID, newProps);
+                this.madeDynamic = true;
+            }
+        };
+
+        this.restoreTargetEntityOriginalProps = function() {
+            if (this.madeDynamic) {
+                var props = {};
+                props.dynamic = false;
```
Comment (SamGondelman) [inline scripts/system/controllers/controllerModules/farActionGrabEntity.js]: I'm not totally familiar with this code or what you're trying to do, but what if the entity was dynamic before?  Won't this always make it non-dynamic?

Comment (druiz17) [reply scripts/system/controllers/controllerModules/farActionGrabEntity.js]: That should only be called is the madeDynamic flag is true.

**Comment by Nex-Pro** [comment]
@druiz17 Sorry I was in HMD I'll start again

**Comment by Nex-Pro** [comment]
Awaiting fresh build

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4716/HighFidelity-Beta-PR11760-c9d67f762bbcd1db95b73faf7f624fd3c8eacd39.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4716/HighFidelity-Beta-PR11760-c9d67f762bbcd1db95b73faf7f624fd3c8eacd39.exe

**Comment by Nex-Pro** [comment]
Retesting

**Comment by Nex-Pro** [comment]
### PASS

**Pull request merged**